app/controllers/tasks_controller.rb

```diff
@@ -0,0 +1,117 @@
+class TasksController < ApplicationController
+    def index
+        @tasks = Task.all.order(:id)
+    end
+
+    def show
+        task_id = params[:id].to_i
+        @task = Task.find_by(id: task_id)
+
+        if task_id < 0
+            flash[:Error] = "Could not find task with id: -1"
+            # redirect_to "Could not find task with id: -1"
+        end
+
+        if @task.nil?
+            redirect_to new_task_path
+            return
+        end
+       
+    end
+
+    def new
+        @task = Task.new
+    end
+
+    def create
+
+        @task = Task.new(task_params) #instantiate a new task using strong params
+            
+            # this code is the same
+        # task_name = params[:name]
+        # task_description = params[:description]
+        # task = Task.create(
+        #     name: task_name,
+        #     description: task_description
+        # )
+
+        # redirect_to :controller => 'tasks', :action => 'index'
+        
+        
+        # @task = Task.new( 
+        #     name: params[:task][:name], 
+        #     description: params[:task][:description],
+        #     completed: params[:task][:completed]
+        # )
+
+            if @task.save
+                redirect_to task_path(@task.id)
+        
+            else
+                render new_task_path
+            end
+    end
+    
+    def edit
+        id = params[:id].to_i
+        if id < 0
+            redirect_to root_path
+        end
+        @task = Task.find_by(id: id )
+        
+    end
+        
+    def update
+        id = params[:id].to_i
+        if id < 0
```

Comment:
Your conditional `if id < 0` is to help determine if the task was not found. However, this is not a very solid way of approaching the problem. This logic works if `params[:id]` is equal to `-1`. What if `params[:id]` is equal to `tacocat`? Then `id` would equal `0`. Also, what if `params[:id]` is equal to `999`, which is a valid number, but there isn't a task with that ID number?

The better thing to do would be to check if `Task.find_by` gives back a task or if it gives back `nil`.

Let me know if you have questions, because this is really important